Extensions/XMPPMUCSUB/XMPPMUCSUB.m

```diff
@@ -0,0 +1,789 @@
+     *         <items node="urn:xmpp:mucsub:nodes:messages">
+     *             <item id="18277869892147515942">
+     *                 <message xmlns="jabber:client" from="coven@muc.shakespeare.example/secondwitch" to="hag66@shakespeare.example/pda" type="groupchat">
+     *                     <archived xmlns="urn:xmpp:mam:tmp" by="muc.shakespeare.example" id="1467896732929849" />
+     *                     <stanza-id xmlns="urn:xmpp:sid:0" by="muc.shakespeare.example" id="1467896732929849" />
+     *                     <body>Hello from the MUC room !</body>
+     *                 </message>
+     *             </item>
+     *        </items>
+     *     </event>
+     * </message>
+     */
+    
+    /*
+     * Presence changes in the MUC room are received wrapped in the same way by subscribers which subscribed to node urn:xmpp:mucsub:nodes:presence:
+     *
+     * <message from="coven@muc.shakespeare.example" to="hag66@shakespeare.example/pda">
+     *     <event xmlns="http://jabber.org/protocol/pubsub#event">
+     *         <items node="urn:xmpp:mucsub:nodes:presences">
+     *             <item id="8170705750417052518">
+     *                 <presence xmlns="jabber:client" from="coven@muc.shakespeare.example/secondwitch" type="unavailable" to="hag66@shakespeare.example/pda">
+     *                     <x xmlns="http://jabber.org/protocol/muc#user">
+     *                         <item affiliation="none" role="none" />
+     *                     </x>
+     *                 </presence>
+     *             </item>
+     *         </items>
+     *     </event>
+     * </message>
+     */
+    
+    /*
+     * If subscriber is subscribed to node urn:xmpp:mucsub:nodes:subscribers, message will be sent for every mucsub subscription change. When a user becomes a subscriber:
+     *
+     * <message from="coven@muc.shakespeare.example" to="hag66@shakespeare.example/pda">
+     *     <event xmlns="http://jabber.org/protocol/pubsub#event">
+     *         <items node="urn:xmpp:mucsub:nodes:subscribers">
+     *             <item id="17895981155977588737">
+     *                 <subscribe xmlns="urn:xmpp:mucsub:0" jid="bob@server.com" nick="bob"/>
+     *             </item>
+     *         </items>
+     *     </event>
+     * </message>
+     */
+    
+    /*
+     * When a user lost its subscription:
+     *
+     * <message from="coven@muc.shakespeare.example" to="hag66@shakespeare.example/pda">
+     *     <event xmlns="http://jabber.org/protocol/pubsub#event">
+     *         <items node="urn:xmpp:mucsub:nodes:subscribers">
+     *             <item id="10776102417321261057">
+     *                 <unsubscribe xmlns="urn:xmpp:mucsub:0" jid="bob@server.com" nick="bob"/>
+     *             </item>
+     *         </items>
+     *     </event>
+     * </message>
+     */
+    
+    NSXMLElement *event = [message elementForName:@"event" xmlns:XMPPPubSubEventNamespace];
```

Comment:
We have a lot of strings being used a lot... can we make them constants? E.g. `kElementEvent`, `kElementItems`, etc...

Reply:
Done